wpi-many.sh

```diff
@@ -0,0 +1,217 @@
+#             on the number of slashes, so excluding https:// is an error.
+#             
+#             If the repository's owner is the user specified by the -u flag,
+#             then each line owned by that user must contain a third element,
+#             the original github repository.
+#
+# The meaning of each optional argument is:
+#
+# -u user : the GitHub owner for repositories that have
+#           been forked and modified. These repositories must have a third entry
+#           in the infile indicating their origin. Default is "$USER".
+#
+# -t timeout : the timeout to use, in seconds
+#
+# After these arguments, any remaining arguments are passed directly
+# to DLJC without modification. See the documentation of DLJC for
+# an explanation of these arguments: https://github.com/kelloggm/do-like-javac
+# At least one such argument is required: --checker, which tells DLJC what
+# typechecker to run.
+#
+
+while getopts "o:i:u:t:" opt; do
+  case $opt in
+    o) OUTDIR="$OPTARG"
+       ;;
+    i) INLIST="$OPTARG"
+       ;;
+    u) GITHUB_USER="$OPTARG"
+       ;;
+    t) TOUT="$OPTARG"
+       ;;        
+    \?) # the remainder of the arguments will be passed to DLJC directly
+       ;;
+  esac
+done
+
+# shift so that the other arguments (that should be passed to dljc) are all
+# that's in $@
+shift $(( OPTIND - 1 ))
+
+# check required arguments and environment variables:
+
+if [ "x${JAVA8_HOME}" = "x" ]; then
+    echo "JAVA8_HOME must be set to a Java 8 JDK"
+    exit 1
+fi
+
+if [ ! -d "${JAVA8_HOME}" ]; then
+    echo "JAVA8_HOME is set to a non-existent directory ${JAVA8_HOME}"
+    exit 1
+fi
+
+if [ "x${JAVA11_HOME}" = "x" ]; then
+    echo "JAVA11_HOME must be set to a Java 11 JDK"
+    exit 1
+fi
+
+if [ ! -d "${JAVA11_HOME}" ]; then
+    echo "JAVA11_HOME is set to a non-existent directory ${JAVA11_HOME}"
+    exit 1
+fi
+
+if [ "x${CHECKERFRAMEWORK}" = "x" ]; then
+    echo "CHECKERFRAMEWORK is not set; it must be set to a locally-built Checker Framework. Please clone and build github.com/typetools/checker-framework"
+    exit 2
+fi
+
+if [ ! -d "${CHECKERFRAMEWORK}" ]; then
+    echo "CHECKERFRAMEWORK is set to a non-existent directory ${CHECKERFRAMEWORK}"
+    exit 2
+fi
+
+if [ "x${OUTDIR}" = "x" ]; then
+    echo "You must specify an output directory using the -o argument."
+    exit 3
+fi
+
+if [ "x${INLIST}" = "x" ]; then
+    echo "You must specify an input file using the -i argument."
+    exit 4
+fi
+
+if [ "x${GITHUB_USER}" = "x" ]; then
+    GITHUB_USER="${USER}"
+fi
+
+### Script
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# clone or update DLJC
+if [ ! -d "${SCRIPTDIR}/../do-like-javac" ]; then
+    git -C "${SCRIPTDIR}/.." clone https://github.com/kelloggm/do-like-javac
+else
+    git -C "${SCRIPTDIR}/../do-like-javac" pull
+fi
+
+export DLJC="${SCRIPTDIR}/../do-like-javac/dljc"
+    
+export PATH="${JAVA_HOME}/bin:${PATH}"
```

Comment:
Should this test that JAVA_HOME is set?  If not, it refers to a real directory `/bin` and may change what executables are found when looking them up on the path.

The documentation for this script does not state that JAVA_HOME must be set.  Can it just be taken from JAVA11_HOME?